### DISCUSSION TITLE
[General] Hardware Acceleration - Intel iGPU ?

### DISCUSSION THREAD

**Discussion by luckylinux** [blakeblackshear/frigate#13624] [General]
I am debating whether to use a System I already have, possibly just buy a CPU with iGPU for another Platform or starting from Scracth using a new Platform for best Performances.

The goal is to have some Hardware Encoding/Decoding Capabilities for 5 ... 10 Cameras, at least 4 of which are 4k/8MP (the other will be 4MP).

I figured that it would be better, both in Terms of Hardware Support (:roll_eyes: ... NVIDIA ... Legacy Cards Support on GNU/Linux :angry:) and Power Consumption to avoid installing a Dedicated GPU for this.

I want to keep all doors opened for Virtualization, so VT-x AND VT-d are a must. ECC RAM is definitively preferred !

Currently Have:
- ASUS P9D WS
- Intel Xeon E3-1245 v3 (with Intel HD Graphics P4600)
- 32 GB DDR3 ECC UDIMM RAM

Considering:
- Supermicro X11SSZ-TLN4F (already have) OR ASUS P10S WS (would need to buy)
- Intel Xeon E3-1275 v6 (with Intel HD Graphics P630)
- 32 GB ... 64 GB DDR4 ECC UDIMM RAM

From Scratch based on Ideas from this [Blog Post](https://mattgadient.com/7-watts-idle-on-intel-12th-13th-gen-the-foundation-for-building-a-low-power-server-nas/):
- ASUS Prime H770-Plus D4
- Intel i5-12400 (with Intel UHD Graphics 730)
- 32 GB ... 64 GB DDR4 non-ECC UDIMM RAM

The drawback with the "From Scratch" approach is that it's quite Expensive. Approx. 140-150 EUR including Shipping for the CPU, approx. 150-200 EUR for the Motherboard plus the RAM. 

Furthermore this is a very feature-less Motherboard, no no Intel VT-D (PCIe Passthrough) at all. Heck, IIRC I didn't even find a mention of "Virtualization" in the Motherboard Manual, so I'm not even sure VT-x is supported. 

Going to a Z690 Motherboard on the other Hand would bump this to 300+ EUR.

Whereas for a Board with the B660, e.g. ASUS PRIME B660-PLUS D4, but also no mention about "Virtualization" in the Manual either. There is a separate "Intel 600 Series - BIOS Manual", but it's NOT clear to what it applies (Bxxx, Hxxx, Zxxxx).

Not sure if there are relevant offers from AMD in that Segment, which would have good Hardware Transcoding Capabilitries in regards to Frigate. I know that AMD APUs used to be quite good, but everybody seems to talk about Intel these Days :roll_eyes:. 

AMD would probably also allow me to use ECC RAM, although I'm not fully sure if also on their APUs all support ECC.

This [AMD Ryzen 7 5700G/5700GE APU](https://www.videocardbenchmark.net/gpu.php?gpu=Ryzen+7+5700GE+with+Radeon+Graphics&id=4433) seems to demolish [Intel i5-12400 iGPU](https://www.videocardbenchmark.net/gpu.php?gpu=Intel+UHD+Graphics+730&id=4379) while performances seem to be approx. the same on the CPU side with a bit better Performance on the AMD side (https://www.cpubenchmark.net/compare/4677vs4323vs4387/Intel-i5-12400-vs-AMD-Ryzen-7-5700G-vs-AMD-Ryzen-7-5700GE).

Maybe a more fair Comparison would be with the AMD Ryzen 5 5600G APU (approx. same Graphics Performance, priced 120 EUR instead of 160 EUR for the 5700G).

But for ECC Support, I'd need either the 5750G or 5650G, both of which seem to score lower Performance in Graphics Tests ([here for 5750G](https://www.videocardbenchmark.net/gpu.php?gpu=Ryzen+7+PRO+5750G+with+Radeon+Graphics&id=4427) and [here for 5650G](https://www.videocardbenchmark.net/gpu.php?gpu=Radeon+Ryzen+5+PRO+5650G&id=4485)). 

But these last ones are much more difficult to get hold on and are more expensive :disappointed:.

Any Thoughts ?

**Comment by NickM-27** [comment]
I'm not sure on the virtualization requirement, I use docker only so it's never been a concern. I can say I have the 5700G and it runs 8 cameras at well as occasional transcoding for jellyfin without using more than 10% of the GPU

**Comment by luckylinux** [comment]
I could get a new AMD 5750G for 233 EUR or the AMD 5650G for 128 EUR. eBay second-hand Prices are not much better and there isn't much availability for these AMD PRO APUs to be honest ...

The Flip side is that I would have 2x32GB ECC DDR4 that doesn't work anywhere else that I could use on another AM4 Platform :rofl:.

**Comment by jekyllus** [comment]
Recently I was looking for new HW for TrueNAS Scale with ECC RAM, but intel LGA1700 desktop chipsets like Bxxx, Hxxx, Zxxx doesn't support ECC RAM. If I good remember for ECC support there are Wxxx chipsets with very expensive boards... :(
Finally I took B760 board with DDR4 and i3-12100 (iGPU) for a minimum prize and invested more to storage redundancy

**Comment by amm009uk** [comment]
Using DELL Optiplex micro 7070 with 32gb ram and i7 9th gen. Its small and quiet. Only 4 cameras and once it settles down its at ~5% CPU.
Running containers for frigate and zerotier. Planning to add 2 more cameras and home assistant.
Cost £200.
